### ISSUE TITLE
Feature: compare api against a snapshot with maven plugin

### ISSUE THREAD

**Issue by surli** [revapi/revapi#54]
Hi,

I'm not sure if the feature is already supported or not, so if it is just close the ticket here and answer on [my stackoverflow question](http://stackoverflow.com/questions/41162349/is-it-possible-to-use-revapi-maven-plugin-to-do-check-against-a-snapshot-version) :)

I want to be able to compare the last build against the last snapshot uploaded on our snapshot repository: the idea is to automatically detect regressions from build to build in our jenkins. However I have the impress that currently revapi only gets the last release and is not able to get last snapshots, or at least I did not find how to do that on the doc.

My current config is the following: 
```
 <plugin>
        <groupId>org.revapi</groupId>
        <artifactId>revapi-maven-plugin</artifactId>
        <version>0.7.0</version>
        <dependencies>
          <dependency>
            <groupId>org.revapi</groupId>
            <artifactId>revapi-java</artifactId>
            <version>0.11.2</version>
          </dependency>
        </dependencies>
        <executions>
          <execution>
            <id>revapi-check</id>
            <goals>
              <goal>check</goal>
            </goals>
          </execution>
        </executions>
        <configuration>
          <alwaysCheckForReleaseVersion>false</alwaysCheckForReleaseVersion>
          <oldVersion>LATEST</oldVersion>
          <analysisConfiguration>
            {
              "revapi": {
                "java": {
                  "filter": {
                    "packages": {
                      "regex": true,
                      "include": ["spoon\\..*"],
                      "exclude": ["spoon\\..*\\.test(\\..*)?"]
                    }
                  }
                }
              }
            }
          </analysisConfiguration>
        </configuration>
      </plugin>
    </plugins>
```

**Comment by metlos** [comment]
You are right that this is not possible at the moment. As I mentioned in the answer on http://stackoverflow.com/a/41233689/1969945 revapi assumes that you compare against the last release, not last build.

Could you help me understand your usecase? What will be the next steps when your CI detects an API change since the last build?

**Comment by surli** [comment]
Thanks again for the answer. The general idea is to be able to check if proposed PR contain API regression by testing the PR with the last built version. 
We did not want to check it with the last release as some choices are made and our API evolve a lot from release to release.
The process should be something like: Release 1 with a given API. Dev on release 2, each night a new snapshot is deployed. PR 1, which broke the API: we discuss if we really want to make those changes in the API, if yes we document it and maybe we decide then if we are going to do a major or minor release. Then when the time is come to do Release 2, we check it with Release 1 and verify that only the documented changes are made in the API.

**Comment by metlos** [comment]
That is actually the workflow that Revapi intends to support 😉 

> The process should be something like: Release 1 with a given API. Dev on release 2, each night a new snapshot is deployed. PR 1, which broke the API: we discuss if we really want to make those changes in the API, if yes we document it and maybe we decide then if we are going to do a major or minor release. Then when the time is come to do Release 2, we check it with Release 1 and verify that only the documented changes are made in the API.

If you document your API changes using Revapi: http://revapi.org/modules/revapi-basic-features/extensions/ignore.html, you don't need to a) do the extra Release1 vs. Release2 check when you release Release2 and b) you cannot run into problems of "stealth" API changes that are "lost" on you when a new snapshot is accidentally built without first resolving the API changes.

I understand though that this workflow can be a little bit tiresome when the API changes a lot.

So to support build-to-build tracking, I need to:
* be able to check "same" versions - `x.y-SNAPHOST` vs. `x.y-SNAPSHOT`
* make sure that the old version is taken from a snapshot repo, not locally built

**Comment by metlos** [comment]
As a side-note - the above mentioned `ignore` extension is going to get some substantial improvements in the next release due to fixing of #17.

**Comment by metlos** [comment]
Oh, and if you read through http://revapi.org/modules/revapi-maven-plugin/examples/multi-file-configuration.html you can see how to set up a single "intended API changes" file for multiple versions so that you have 1 place to look for all API changes in all releases.

**Comment by surli** [comment]
> So to support build-to-build tracking, I need to:
> -  be able to check "same" versions - x.y-SNAPHOST vs. x.y-SNAPSHOT
> - make sure that the old version is taken from a snapshot repo, not locally built

Yes this is exactly the feature I need. 
Thanks for the extra information!

**Comment by metlos** [comment]
This will be available in `revapi-maven-plugin-0.8.0` that I hope to release soon. Thanks for the idea!
